tests/5.0/atomic/test_atomic_num_hint.F90

```diff
@@ -0,0 +1,80 @@
+!//===------ test_atomic_num_hint.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+!  This test checks if atomic hints passed by enum value
+!  are accepted by the compiler. If the sync hint is not
+!  yet defined in the specification, it defaults to 
+!  omp_sync_hint_none (0x0). 
+!
+!//===---------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_atomic_num_hint
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_VERBOSE(test_atomic_with_used_enum_value() .ne. 0)
+  OMPVV_TEST_VERBOSE(test_atomic_with_unused_enum_value() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION test_atomic_with_used_enum_value()
+    INTEGER:: x, num_threads, errors
+
+    errors = 0
+    x = 0
+    num_threads = -1
+
+    OMPVV_INFOMSG("test_atomic_with_used_enum_value")
+
+    !$omp parallel num_threads(2) default(shared)
+       IF (omp_get_thread_num() .EQ. 0) THEN
+          num_threads = omp_get_num_threads()
+       END IF
+       !$omp atomic hint(4) ! corrosponds to omp_sync_hint_nonspeculative
```

Comment:
```suggestion
       !$omp atomic hint(4) ! corresponds to omp_sync_hint_nonspeculative
```
Question: Why does the testcase use '4' instead of omp_sync_hint_nonspeculative ? That values should be in 'omp.h' (for C/C++) and in 'omp_lib.h' (Fortran include) and in the 'omp_lib' module (Fortran).  [Same for the C testcase.]

As the value is normative, using '4' should also work – but it feels much more natural to use the name, doesn't it? And if 4 works and omp_sync_hint_nonspeculative fails, that would be IMHO an implementation bug.

Reply:
We have a separate test `test_atomic_hint.c` and `test_atomic_hint.F90` where hint constants defined by spec are used. The purpose of this test is to see whether implementations work if hint clause argument is an integer number.